CODE-GUIDELINES.md

```diff
@@ -37,3 +37,35 @@ test('...', () => {
   vi.mocked(window.windowMethod).mockResolvedValue('a string');
 });
 ```
+
+### Do not use `try {} catch` in tests
+
+Let's have an async function such as
+
+```ts
+export function foo(): Promise<void> {
+  throw new Error('bar');
+}
+```
+
+Let's say we want to check that under certain condition the function throw an error (here always), we _could_ write the following code
+
+```ts
+// bad
+test('expect error to be properly catched and verified', async () => {
+  try {
+    await foo();
+  } catch(err: unknown) {
```

Comment:
Can be probably ignored since we want to make it simple, 
but can't we use instead of `undefined` something like `Error`? Won't the linter yell at you that the undefined does not have a message attribute? (Thinking of creating an interface Error, or something like that). 

`err && typeof err === 'object' && 'message' in err && typeof err.message === 'string'`

Reply:
maybe we just do `expect(String(err)).contains('bar')` or something else because I think the idea is to say that we should not use expect in a catch clause

It might be overkill to use all the syntax check to verify if err has a field message just for this purpose. I agree it's not compiling on the current code but maybe we can have a in-the-middle solution where we can still have something short and with a valid typechecking